image/distributionutil/login.go

```diff
@@ -41,7 +46,21 @@ func Login(ctx context.Context, authConfig *types.AuthConfig) error {
 	}
 
 	modifiers := dockerRegistry.Headers(dockerversion.DockerUserAgent(ctx), nil)
-	authTransport := transport.NewTransport(dockerRegistry.NewTransport(nil), modifiers...)
+	var tlsc *tls.Config
+	if dockerCertPath := os.Getenv("DOCKER_CERT_PATH"); dockerCertPath != "" {
+		options := tlsconfig.Options{
+			CAFile:   filepath.Join(dockerCertPath, "ca.pem"),
+			CertFile: filepath.Join(dockerCertPath, "cert.pem"),
+			KeyFile:  filepath.Join(dockerCertPath, "key.pem"),
+		}
+		tlsc, err = tlsconfig.Client(options)
+		if err != nil {
+			return err
+		}
+	}
+	base := dockerRegistry.NewTransport(tlsc)
+	base.TLSClientConfig.InsecureSkipVerify = os.Getenv("SKIP_TLS_VERIFY") == "true"
```

Comment:
It's good, but do it before getting "DOCKER_CERT_PATH". Because if user decides to skip tls verify, why do we fetch cert?
Or skip tls and retrieve cert are not the same thing to do ?

Reply:
So user must set SKIP_TLS_VERIFY env to skip the verify issue? We can show some log if the error occured